test/unit/master/test_cluster_master.py

```diff
@@ -57,6 +58,45 @@ def test_get_slave_returns_expected_value_given_valid_arguments(self):
         self.assertEqual('leonardo.turtles.gov', actual_slave_by_url.url,
                          'Retrieved slave should have the same url as requested.')
 
+    def test_connect_slave_adds_new_slave_if_slave_never_connected_before(self):
+        master = ClusterMaster()
+
+        master.connect_slave('never-before-seen.turtles.gov', 10)
+
+        self.assertEqual(1, len(master.all_slaves_by_id()), 'Exactly one slave should be registered with the master.')
+        self.assertIsNotNone(master.get_slave(slave_id=None, slave_url='never-before-seen.turtles.gov'),
+                             'Registered slave does not have the expected url.')
+
+    def test_connect_slave_with_existing_dead_slave_marks_it_as_alive(self):
+        master = ClusterMaster()
```

Comment:
The 'arrange' part of these three tests can be shared somewhat to reduce the number of lines.